students/brian_warn/session04/kata_test.py

```diff
@@ -0,0 +1,94 @@
+#!/Library/Frameworks/Python.framework/Versions/3.6/bin/python
+# Kata Fourteen
+import random
+
+
+def main():
+    exceptions = ['.', '?', ' ', '!', ',', '-']
+
+    def build_trigram(infile):
+        ''' Function that builds the trigram dictionary from the source file. '''
+
+        trigrams = {}
+        with open(infile) as sherlock:
+            sherlock.readline()
+            for line in sherlock:
+                line = line.rstrip('\r\n')
+                word = line.split(" ")
+                for i in range(len(word) - 2):
+                    pair = tuple(word[i:i + 2])
+                    follower = word[i + 2]
+                    if follower not in exceptions:
+                        pass
```

Comment:
this does nothing at all -- is that because you are holding the place until you decide what to put there?

or did yo intend to put "continue" in there, which would skip the rest of the code and continue the loop at the next pair...

careful with that though, the "exeption" would then show up in the next pair...

I'm not totally sure where you are heading here but if you want to ignore the punctuation, better to removeit from the text before running this loop.

Reply:
I was still sorting out what to do here.  I think your `setdefault` approach is the direction I need to take.